profiles/common/modules/features/nexteuropa_newsroom/classes/NewsroomAgenda.inc

```diff
@@ -0,0 +1,679 @@
+  }
+
+  /**
+   * Returns agenda navigation URL.
+   *
+   * @param DateTime $date
+   *   Datetime object.
+   *
+   * @return string
+   *   URL string.
+   */
+  public function getNavigationUrl(DateTime $date) {
+    $this->clearUrlParts();
+    $this->addUrlPart($this->getView());
+    $this->addUrlPart($this->getTypeUrl());
+    $this->addUrlPart($this->getTopicUrl());
+    $this->setUrlItems($date);
+    return 'newsroom-agenda/' . implode('/', $this->getUrlParts());
+  }
+
+  /**
+   * Setups agenda conditions.
+   */
+  protected function setupAgenda() {
+    $this->setCurrentItems($this->buildAgenda($this->getCurrentDate(), $this->getNextDate()));
+  }
+
+  /**
+   * Loads result from the current query.
+   *
+   * @return array
+   *   Result.
+   */
+  protected function loadResult() {
+    $end_date_condition = db_and()
+      ->condition('d.field_newsroom_item_date_value2', $this->getCurrentDate()->format('Y-m-d'), '>=')
+      ->condition('d.field_newsroom_item_date_value', $this->getNextDate()->format('Y-m-d'), '<=');
+
+    $start_date_condition = db_and()
+      ->condition('d.field_newsroom_item_date_value', $this->getCurrentDate()->format('Y-m-d'), '>=')
+      ->condition('d.field_newsroom_item_date_value', $this->getNextDate()->format('Y-m-d'), '<=')
+      ->isNull('d.field_newsroom_item_date_value2');
+
+    $date_condition = db_or();
+    $date_condition->condition($end_date_condition);
+    $date_condition->condition($start_date_condition);
+
+    $this->getQuery()->condition($date_condition);
+    return $this->getQuery()->execute()->fetchAll();
+  }
+
+  /**
+   * Sort agenda items.
+   */
+  protected function sortItems($items) {
+    foreach ($items as $key => $item) {
+      $events = $item->getItems();
+      usort($events, function ($a, $b) {
+        $a_start_date_obj = $a->start_date_obj;
+        $b_start_date_obj = $b->start_date_obj;
+        $a_end_date_obj = $a->end_date_obj;
+        $b_end_date_obj = $b->end_date_obj;
+
+        if ($a_end_date_obj < $b_end_date_obj) {
+          return -1;
+        }
+        elseif ($a_end_date_obj > $b_end_date_obj) {
+          return 1;
+        }
+        else {
+          if ($a_end_date_obj == $b_end_date_obj) {
+            return 0;
+          }
+          else {
+            return $a_start_date_obj > $b_start_date_obj ? -1 : 1;
+          }
+        }
+      });
+      $item->setItems($events);
+      $items[$key] = $item;
+    }
+
+    return $items;
+  }
+
+  /**
+   * Get IDs array from topics array.
+   *
+   * @return array
+   *   Array of IDs.
+   */
+  protected function getTopicsId() {
+    $topic_ids = array();
+    if (count($this->getTopics()) > 0) {
+      foreach ($this->getTopics() as $topic) {
+        $topic_ids[] = $topic->tid;
+      }
+    }
+    return $topic_ids;
+  }
+
+  /**
+   * Get IDs array from topics array.
+   *
+   * @return array
+   *   Array of IDs.
+   */
+  protected function getResultTopicsId() {
```

Comment:
I have checked it. I have increased it, but I cannot change it permanent, because we import the items everyday and new topics can be associated with items, In this case the result from the cache will irrelevant.